dbms/src/Storages/MergeTree/MergeTreeDataPartTTLInfo.cpp

```diff
@@ -82,6 +102,26 @@ void MergeTreeDataPartTTLInfos::write(WriteBuffer & out) const
         writeIntText(table_ttl.max, out);
         writeString("}", out);
     }
+    if (!moves_ttl.empty())
+    {
+        if (!columns_ttl.empty() || table_ttl.min)
+            writeString(",", out);
+        writeString("\"moves\":[", out);
+        for (auto it = moves_ttl.begin(); it != moves_ttl.end(); ++it)
+        {
+            if (it != moves_ttl.begin())
+                writeString(",", out);
+
+            writeString("{\"expression\":", out);
```

Comment:
Why we store expression here? It should be stored in table definition.

Reply:
This thing is basically an additional protection level, to not make wrong moves. Expressions from here are not being analysed for anything but rather used by keys for min/max values. If someone ALTER TTL expressions, data will not be moved according to stale TTLs.